src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java

```diff
@@ -243,10 +243,26 @@ private static String joinPaths(String prefix, String suffix) {
         return prefix + '/' + suffix;
     }
 
-    public synchronized void terminateJenkinsSlave(String name) {
-        LOGGER.info("Terminating jenkins slave " + name);
-
-        TaskID taskId = TaskID.newBuilder().setValue(name).build();
+  /**
+   * When using the Docker Image Can Be Customized option you can specify the docker image to use for example
+   * foo/bar:version1 which is appended at the end of the mesos task name (then limited to 64 chars)
+   *
+   * This is a problem because mesos task names cannot contain "/" as the mesos task name is used in
+   * the directory structure to run the task
+   * It is also a problem for the docker hostname which cannot contain ":" or "/"
+   * See RFC 952
+   *
+   * @param name the task or hostname
+   * @return cleaned up hostname or task name where invalid chars are replaced by the "_" character
+   */
+  private static String replaceInvalidHostnameCharacters(String name) {
```

Comment:
Instead of fixing the task name, why dont we fix the slave name instead? That would be change in one place (MesosUtils.buildNodeName()) which I think is cleaner.

Reply:
I'll be happy to merge this once comments get resolved. Thanks for fixing the issue and sorry it took so long to get feedback.

Reply:
would changing the slave name ripple through to the task name?

Reply:
yes, they are the same.